clippy_lints/src/allow_attribute.rs

```diff
@@ -0,0 +1,93 @@
+use ast::{AttrStyle, MetaItemKind};
+use clippy_utils::{diagnostics::span_lint_and_sugg, source::snippet};
+use rustc_ast as ast;
+use rustc_errors::Applicability;
+use rustc_lint::{LateContext, LateLintPass};
+use rustc_session::{declare_tool_lint, impl_lint_pass};
+use rustc_span::{symbol::Ident, BytePos};
+
+declare_clippy_lint! {
+    /// ### What it does
+    /// Detects uses of the `#[allow]` attribute and suggests to replace it with the new `#[expect]` attribute implemented by `#![feature(lint_reasons)]` ([RFC 2383](https://rust-lang.github.io/rfcs/2383-lint-reasons.html))
+    /// ### Why is this bad?
+    /// Using `#[allow]` isn't bad, but `#[expect]` may be preferred as it lints if the code **doesn't** produce a warning.
+    /// ### Example
+    /// ```rust,ignore
```

Comment:
Do we need the ignore here? :)

Reply:
If `,ignore` is removed CI will fail because it should be:

```rust
 #[allow(unused_mut)]
 fn foo() -> usize {
    let mut a: Vec<TYPE> = Vec::new();
    a.len()
}
```

I consider that adding that type annotation doesn't help showing what the lint does, and clutters the example making it harder to understand these examples.

Reply:
Makes sense, you could just use `vec![1]` to initialize it, that would give it the default integer type :). I prefer the example to be tested, but it's not necessary :)